lib/node-token-validation/src/cache/NodeCacheManager.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthorityMetadataEntity, CacheManager, ICrypto, Logger, ValidCacheType } from "@azure/msal-common";
+import { CacheKVStore } from "./CacheTypes";
+
+/**
+ * This class class implements storage for node-token-validation.
+ */
+export class NodeCacheManager extends CacheManager {
+    private logger: Logger;
+    private cache: CacheKVStore = new Map();
+    private changeEmitters: Array<Function> = [];
+
+    constructor(logger: Logger, clientId: string, cryptoImpl: ICrypto) {
+        super(clientId, cryptoImpl);
+        this.logger = logger;
+    }
+
+    /**
+     * Queue up callbacks
+     * @param func A callback function for cache change indication
+     */
+    registerChangeEmitter(func: () => void): void {
+        this.changeEmitters.push(func);
+    }
+    
+    /**
+     * Invoke the callback when cache changes
+     */
+    emitChange(): void {
+        this.changeEmitters.forEach(func => func.call(null));
+    }
+
+    /**
+     * Gets the current cache key-value store
+     * @returns {CacheKVStore} Key Value store for in-memory storage
+     */
+    getCache(): CacheKVStore {
+        this.logger.trace("Getting cache");
+        return this.cache;
+    }
+
+    /**
+     * Sets the current cache key-value store
+     * @param {CacheKVStore} cache 
+     */
+    setCache(cache: CacheKVStore): void {
+        this.logger.trace("Setting cache");
+        this.cache = cache;
+
+        // Mark change in cache
+        this.emitChange();
```

Comment:
Why would you need this? Ideally this was set as an event marker to denote change in the cache to enable disk read/write with data integrity for `msal-node-extensions` where we needed to read from an SSO island (where multiple apps have access to the token store - MSAL .Net, Py, Java etc..).

Would we need similar storage choices for `node-token-validation`?

Happy to talk if you need more context.

Reply:
Thanks for the chat today. We concluded that this event marker was not needed for `node-token-validation`, and it has been removed.